drivers/xen/gntdev-dmabuf.c

```diff
@@ -10,20 +10,27 @@
 
 #include <linux/kernel.h>
 #include <linux/errno.h>
+#include <linux/delay.h>
 #include <linux/dma-buf.h>
+#include <linux/dma-resv.h>
 #include <linux/slab.h>
 #include <linux/types.h>
 #include <linux/uaccess.h>
 #include <linux/module.h>
+#include <linux/notifier.h>
 
 #include <xen/xen.h>
 #include <xen/grant_table.h>
+#include <xen/mem-reservation.h>
 
 #include "gntdev-common.h"
 #include "gntdev-dmabuf.h"
 
 MODULE_IMPORT_NS(DMA_BUF);
 
+#define RELEASE_FLAG_SYNC	0
+#define RELEASE_FLAG_ASYNC	1
```

Comment:
I see your intention there, but from code standpoint it looks unnatural. Flags usually are defined as bits:

```
#define RELEASE_FLAG_SYNC BIT(0)
```
Which usually implies that async release is denoted by absence of  `RELEASE_FLAG_SYNC`.

If you want to enumerate all possible variants, you could use `enum`

```
enum release_type {
 RELEASE_SYNC = 0,
 RELEASE_ASYNC = 1,
};
```

**BUT**. As `dmabuf_map_release` deals only with sync/async option, you don't need flags and enums. You need just `bool`:

`static int dmabuf_map_release(struct gntdev_dmabuf *gntdev_dmabuf, bool sync);`

I understand that you wanted to make it future-proof, but we don't know the future and we don't know what other options might arise. Maybe you will need another flag, maybe - another option in `enum`, so there is no sense choosing something more complex over just plain old `bool`. Anyways, this function is not even part of internal kernel API, it is local for this module. In the future you always can change its declaration to something that suits you more.

Reply:
Yes. I think you're right. Changed as you suggested